codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ServiceGenerator.java

```diff
@@ -294,6 +316,71 @@ private void generateApplicationProtocolTypes() {
         }).write("");
     }
 
+    private void writeProtocolResolvers() {
```

Comment:
nit: `writeProtocolResolvers` and `generateProtocolResolvers` sound like the same thing. it looks like the former is more writing/generating the invocations and the latter is doing the definitions, so maybe work those intentions into the name of the functions